gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Dataset.java

```diff
@@ -235,4 +219,39 @@ public Table create(String table, TableDefinition definition, BigQuery.TableOpti
   public BigQuery bigquery() {
     return bigquery;
   }
+
+  static Builder builder(BigQuery bigquery, DatasetId datasetId) {
+    return new Builder(bigquery).datasetId(datasetId);
```

Comment:
Nit (fine as is), but at this point we can change the constructor to get also `datasetId` and replace DatasetInfo.Builder empty constructor to one that accepts datasetId.